lib/rucio/api/quarantined_replica.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from rucio.api import permission
+from rucio.db.sqla.session import transactional_session
+from rucio.core.rse import get_rse_id
+from rucio.core.quarantined_replica import add_quarantined_replicas
+from rucio.common.types import InternalScope
+from rucio.common import exception
+
+
+@transactional_session
+def quarantine_file_replicas(replicas: object, issuer: object, rse: object = None, rse_id: object = None, vo: object = 'def', session: object = None) -> object:
+    """
+    Quarantine replicas.
+
+    :param replicas: Either a list of paths (string) or a list of replica infos: {'scope': <scope> (optional), 'name': <name> (optional), 'path':<path>}.
```

Comment:
Why do we allow a list of either strings or dicts here?
Isn't a list of dicts with some optional values enough?

If it is really necessary, I'd suggest splitting this into two functions:
```python
quarantine_file_replicas_string_list(replicas, ...):
    replicas = [{'path': p} for p in replicas]
    quarantine_file_replicas(replicas, ...)

quarantine_file_replicas(replicas, ...): 
```
But only if it is **really** needed!